Grammar/Grammar

```diff
@@ -133,13 +134,15 @@ arglist: argument (',' argument)*  [',']
 # Illegal combinations and orderings are blocked in ast.c:
 # multiple (test comp_for) arguments are blocked; keyword unpackings
 # that precede iterable unpackings are blocked; etc.
-argument: ( test [comp_for] |
+argument: ( test [comp_for_or_async] |
             test '=' test |
             '**' test |
             '*' test )
 
-comp_iter: comp_for | comp_if
-comp_for: [ASYNC] 'for' exprlist 'in' or_test [comp_iter]
+comp_for_or_async: comp_async_for | comp_for
+comp_iter: comp_async_for | comp_for | comp_if
+comp_async_for: 'async' 'for' exprlist 'in' or_test [comp_iter]
+comp_for: 'for' exprlist 'in' or_test [comp_iter]
 comp_if: 'if' test_nocond [comp_iter]
```

Comment:
Why did you do all these renames and added `atom_expr`? Why don't we just replace `ASYNC` with `'async'` (ditto for await)?  I'd avoid making **any** changes to the grammar besides token adjustments.